src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,56 @@
+package tasks;
+
+import java.io.Serializable;
+
+/**
+ * Class that represents a task.
+ * These tasks are added to the task list.
+ */
+public class Task implements Serializable {
+    protected final String description;
+    protected boolean isDone = false;
+
+    /**
+     * Constructor for a Task object.
+     * This is only called by inherited classes to set the task description.
+     * @param description Description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+    }
+
+    /**
+     * Returns a task's done or undone status.
+     * @return Either [X] or [ ] for done and undone respectively.
+     */
+    public String getStatusIcon() {
+        return isDone ? "[X] " : "[ ] ";
+    }
+
+    /**
+     * Marks a task as done.
+     */
+    public void setDone() {
+        isDone = true;
+    }
+
+    /**
+     * Marks a task as undone.
+     */
+    public void setUndone() {
+        isDone = false;
+    }
+
+    /**
+     * Returns the task description.
+     * @return The task description.
+     */
+    @Override
```

Comment:
Both methods are doing the same thing it seems, perhaps you can remove one and just stick with toString.